CMakeLists.txt

```diff
@@ -52,6 +52,17 @@ if (NOT BUILD_NUMBER)
   set(BUILD_NUMBER "0")
 endif ()
 
+# Set the architecture. We do this before the compiler settings, since some
+# of them are arch specific.
+if (CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
+    # TODO: Right now assume it's Intel+SGX for x86_64 processors
+    set(OE_SGX 1)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm.*|ARM.*|aarch64.*|AARCH64.*")
+    set(OE_TRUSTZONE 1)
+else()
+    message(FATAL_ERROR "Unknown processor. Only Intel SGX and ARM TrustZone are supported")
```

Comment:
Silly style comment, but I have been trying to get us consistently down to 2 spaces instead of 4. It was mixed when I encountered it, and I chose the former but don't have a great way to re-indent all the files. You don't have to change this, just wanted to let you know.

Reply:
I'll change it to 2 spaces for this file only to be consistent.